pkg/virt-handler/migration-proxy/migration-proxy.go

```diff
@@ -214,6 +216,10 @@ func (m *migrationProxyManager) StartSourceListener(key string, targetAddress st
 	m.managerLock.Lock()
 	defer m.managerLock.Unlock()
 
+	if netutils.IsIPv6String(targetAddress) {
```

Comment:
@AlonaKaplan 
@EdDev 
for the 2nd PR this function is mandatory of course because we do use there ::1
which is different notation of 127.0.0.1 as 
IPv4 is subset of IPv6

i rechecked there that its indeed mandatory and it is
according the decisions on this PR
i will know if to understand all the comments and where
but first lets please focus on those questions

Reply:
about the first comment

it cant, because line 223 creates a lookup table with targetAddress, 
and later on the key is the formatted targetAddress, which must be as the one that was used
to build that table.
moving it will corrupt the logic, i tried that in the past.

Reply:
The `[]` wrapper is an address formatting that the netutils package requires for working with sockets.
On the other hand, the lookup check is about the IP identifier, which does not have to require those braces (as it does not work directly with the netutils). The lookup seems independent in this regard and I see no reason to enforce using braces there.
I guess the other option would have been to require in the API to represent the IPv6 address in the correct format in the first place, but I am not sure about the implications of this. We could explore it in the future.

At the moment I think we should do the minimum needed without affecting anything else, i.e. normalize the IPv6 address where is it a must.

Is something breaking if you move this very close to the listener?

Reply:
yes it breaks, because the key need always to be with [ ] or without [ ]

Reply:
> yes it breaks, because the key need always to be with [ ] or without [ ]
> consistency,
> it means to have two instance, one with and one without
> i think its not needed, the key imo can be with [ ]
> its just a key in just this function, but whatever you want

I'm trying to understand how come the address may be added or checked against one with braces and once without.

The problem with normalizing in the middle is that there is a good chance to miss things or that someone that adds another path int he future will also add these braces again. Therefore, the solution usually is to do it either in the first place it is read or as late as possible.

Reply:
keep in mind please, this is not part of this PR, can we resolve, and i will copy the link to the PR that do matter about it ?
so it wont cause noise here
i already removed the code from this PR

Reply:
first place it is read is 
https://github.com/kubevirt/kubevirt/pull/3392/files#diff-e88e0fbecb2c3ab1c8c08d12a1c8b4f6R243
it must be consistent with 
https://github.com/kubevirt/kubevirt/pull/3392/files#diff-e88e0fbecb2c3ab1c8c08d12a1c8b4f6R257

in line 243 (the first link), it use the instance that was created by line 257 (2nd link)
in the previous run

so maybe should we move it just above 242 

i always look in the line numbers of the right

Reply:
we can use 
https://golang.org/pkg/net/#JoinHostPort
and normalize in 2 places
line below 
https://github.com/kubevirt/kubevirt/pull/3392/files#diff-e88e0fbecb2c3ab1c8c08d12a1c8b4f6R229
and line below
https://github.com/kubevirt/kubevirt/pull/3392/files#diff-e88e0fbecb2c3ab1c8c08d12a1c8b4f6R257

it takes ip:port and makes [ip]:port , which is that we need to "listen" 
and what it saves in the proxy map, which is later must be the key we used to lookup

Reply:
The location is not ideal but it seems that there is a deeper problem with moving this down the stack and close to the net.Listen call.

We should investigate how to make this normalization immediately before the net.Listen call and make sure we are consistent in the code flow with one format.
In any case, this can be handled in a separate PR as it has the potential of touching other areas of the code.